pkg/query-service/rules/base_rule.go

```diff
@@ -131,21 +133,23 @@ func NewBaseRule(id string, orgID valuer.UUID, p *ruletypes.PostableRule, reader
 	}
 
 	baseRule := &BaseRule{
-		id:                id,
-		orgID:             orgID,
-		name:              p.AlertName,
-		source:            p.Source,
-		typ:               p.AlertType,
-		ruleCondition:     p.RuleCondition,
-		evalWindow:        time.Duration(p.EvalWindow),
-		labels:            qslabels.FromMap(p.Labels),
-		annotations:       qslabels.FromMap(p.Annotations),
-		preferredChannels: p.PreferredChannels,
-		health:            ruletypes.HealthUnknown,
-		Active:            map[uint64]*ruletypes.Alert{},
-		reader:            reader,
-		TemporalityMap:    make(map[string]map[v3.Temporality]bool),
-		Threshold:         threshold,
+		id:                  id,
+		orgID:               orgID,
+		name:                p.AlertName,
+		source:              p.Source,
+		typ:                 p.AlertType,
+		ruleCondition:       p.RuleCondition,
+		evalWindow:          time.Duration(p.EvalWindow),
+		labels:              qslabels.FromMap(p.Labels),
+		annotations:         qslabels.FromMap(p.Annotations),
+		preferredChannels:   p.PreferredChannels,
+		health:              ruletypes.HealthUnknown,
+		Active:              map[uint64]*ruletypes.Alert{},
+		reader:              reader,
+		TemporalityMap:      make(map[string]map[v3.Temporality]bool),
+		Threshold:           threshold,
+		NotificationGroupBy: p.NotificationGroups,
```

Comment:
1. if it's a manager use `NotificationManager`
2. if it's a list of group by labels `NotificationGroupBy`

I don't want to see any other use because it's real pain to follow which is which.